cpp/unit/ability.cpp

```diff
@@ -108,7 +132,9 @@ void UngarrisonAbility::invoke(Unit &to_modify, const Command &cmd, bool play_so
 	if (play_sound && this->sound) {
 		this->sound->play();
 	}
-	to_modify.push_action(std::make_unique<UngarrisonAction>(&to_modify, cmd.position()));
+
+	// add as secondary, so primary action is not disrupted
+	to_modify.secondary_action(std::make_unique<UngarrisonAction>(&to_modify, cmd.position()));
```

Comment:
I don't understand why this secondary_action is needed. Why can't we just insert the UngarrisonAction in front of the "primary stack" ?